runtime/monovm-bridge.m

```diff
@@ -392,11 +394,48 @@
 
 #if DOTNET
 
+static void
+xamarin_runtime_config_cleanup (MonovmRuntimeConfigArguments *args, void *user_data)
+{
+	free ((char *) args->runtimeconfig.name.path);
+	free (args);
+}
+
+static void
+xamarin_initialize_runtime_config ()
+{
+	if (xamarin_runtime_configuration_name == NULL) {
+		LOG (PRODUCT ": No runtime config file provided at build time.\n");
+		return;
+	}
+
+	const char *app_path = xamarin_get_bundle_path ();
+	char path [1024];
+	if (!xamarin_locate_assembly_resource_for_root (app_path, NULL, xamarin_runtime_configuration_name, path, sizeof (path))) {
+		LOG (PRODUCT ": Could not locate the runtime config file '%s' in the app bundle: %s\n", xamarin_runtime_configuration_name, app_path);
+		return;
+	}
+
+	MonovmRuntimeConfigArguments *args = (MonovmRuntimeConfigArguments *) calloc (sizeof (MonovmRuntimeConfigArguments), 1);
+	args->kind = 0; // Path of runtimeconfig.blob
+	args->runtimeconfig.name.path = strdup (path);
+
+	int rv = monovm_runtimeconfig_initialize (args, xamarin_runtime_config_cleanup, NULL);
+	if (rv != 0) {
+		LOG_MONOVM (PRODUCT ": Failed to load the runtime config file %s: %i\n", path, rv);
+		return;
+	}
+
+	LOG_MONOVM (PRODUCT ": Loaded the runtime config file %s\n", path);
+}
+
 bool
 xamarin_bridge_vm_initialize (int propertyCount, const char **propertyKeys, const char **propertyValues)
 {
 	int rv;
 
+	xamarin_initialize_runtime_config ();
+
 	rv = monovm_initialize (propertyCount, propertyKeys, propertyValues);
```

Comment:
does not this call reset the values of the config file ?

Reply:
No, the values we pass to the task here:

https://github.com/xamarin/xamarin-macios/pull/11887/files#diff-2e24f1b6f44df7580244d00f106fa1a9e2257536fbca9d7bbdc74571549324a9R522-R524

list the values we will later pass to `monovm_initialize`, so that the task can make sure there are no duplicates.

This is documented here: https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/docs/design/mono/mobile-runtimeconfig-json.md#design-overview